### PR TITLE
Moved model types in datalake service into Models namespace.

### PR DESCRIPTION
**Pull request by katmsft**

### CODE REVIEW

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp [hunk 1]
```diff
@@ -98,14 +98,14 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
      * @brief Create a directory. By default, the destination is overwritten and
      *        if the destination already exists and has a lease the lease is broken.
      * @param options Optional parameters to create the directory the path points to.
-     * @return Azure::Core::Response<CreateDirectoryResult> containing the information of the
-     * created directory
+     * @return Azure::Core::Response<Models::CreateDirectoryResult> containing the information of
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp:101]: The namespaces here are inconsistent.

shouldn't it be `Azure::Core::Response<Azure::Storage::Models::CreateDirectoryResult>`?
Why did you omit `Azure::Storage::`?

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp [hunk 2]
```diff
@@ -149,12 +149,13 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
...
+     * @return Azure::Core::Response<Models::SetDirectoryAccessControlRecursiveResult>
      * @remark This request is sent to dfs endpoint.
      */
-    Azure::Core::Response<SetDirectoryAccessControlRecursiveResult> SetAccessControlRecursive(
-        PathSetAccessControlRecursiveMode mode,
-        std::vector<Acl> acls,
+    Azure::Core::Response<Models::SetDirectoryAccessControlRecursiveResult>
+    SetAccessControlRecursive(
+        Models::PathSetAccessControlRecursiveMode mode,
+        std::vector<Models::Acl> acls,
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp:158]: Are you sure you re-formatted with clang-format?

**Review by Jinming-Hu — APPROVED** [review]

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp (hunk 1 above)
Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp:101]: The full namespace is `Azure::Storage::Files::DataLake::Models::CreateDirectoryResult`, this line is already under `Azure::Storage::Files::DataLake::` so it makes sense to omit what's before `Models` namespace.

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp (hunk 2 above)
Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp:158]: Yes, can show you offline if you think is neccessary.

**Pull request merged**